### PR TITLE
CAPI-1125 : Reading service metadata by service Id

### PR DESCRIPTION
**Pull request by SiniWilson**
### Requirements

[CAPI-1125](https://tmobile.atlassian.net/browse/CAPI-1125)  : Reading service metadata from service id

### Description of the Change

  Service metadata is loaded by the service id in the deployment-env.yml file.
 Changes in writing serverless.yml file

### Benefits

User no longer able to see the service metadata information

### CODE REVIEW

File: jazz-build-module/service-metadata-loader.groovy [hunk 1]
```diff
@@ -66,6 +66,27 @@ def loadServiceMetaData() {
 	}
 }
 
+/**
+ * Load the service metadata from Catalog
+ *
+ */
+ 
+def loadServiceMetaDataByServiceId() {
```
Comment (suryajak) [inline jazz-build-module/service-metadata-loader.groovy]: The signature for this method seems wrong. We should pass serviceId as a param. Once you add that as param, update the method to say loadServiceMetaData

File: delete-serverless-service-build-pack/Jenkinsfile [hunk 2]
```diff
@@ -54,6 +54,7 @@ node {
 	def bitbucket_repo_name
 	def tracking_id
 	def config
```
Comment (suryajak) [inline delete-serverless-service-build-pack/Jenkinsfile]: Do we need both config and configObj?

Comment (suryajak) [reply delete-serverless-service-build-pack/Jenkinsfile]: Looking at the rest of your code, looks like configObj is your temp object for holding data from service module. So I would do 2 things: move the scope of this variable closer there. Also change the name to serviceObj

File: delete-serverless-service-build-pack/Jenkinsfile [hunk 3]
```diff
@@ -144,20 +160,20 @@ node {
 		switch (flowType) {
 			case "API":
 				stage('Undeploy Service') {
-					updateServiceNameConfig(domain, service_name, "dev")
+					updateServiceNameConfig(domain, service_name, "dev",config)
```
Comment (suryajak) [inline delete-serverless-service-build-pack/Jenkinsfile]: Shouldn't we update the signature for updateServiceNameConfig to take only config as config will contain domain and service_name too?

File: jenkins-build-pack-api/Jenkinsfile [hunk 4]
```diff
@@ -92,9 +92,11 @@ node() {
 
 
 	def configPackBBURL = "http://" + var_bitbucket_base + "/scm/" + var_repo_core + "/serverless-config-pack.git"
+	def jazzBuildModuleURL = "http://${var_bitbucket_base}/scm/"+ var_repo_core +"/jazz-build-module.git"
 
 	def config
-
+	def configObj
```
Comment (suryajak) [inline jenkins-build-pack-api/Jenkinsfile]: Similar comments as in delete-serverless-service-build-pack/Jenkinsfile

File: jenkins-build-pack-lambda/Jenkinsfile [hunk 5]
```diff
@@ -99,6 +100,7 @@ node ()  {
 	def bitbucket_name = params.service_name
 	def var_bitbucket_framework_url = "http://${var_bitbucket_base}/scm/${var_repo_loc}/"
 	def config
+	def configObj
```
Comment (suryajak) [inline jenkins-build-pack-lambda/Jenkinsfile]: Similar comments as in delete-serverless-service-build-pack/Jenkinsfile

File: jenkins-build-pack-website/Jenkinsfile [hunk 6]
```diff
@@ -83,6 +84,7 @@ node ()  {
 	def var_bitbucket_framework_url = "http://${var_bitbucket_base}/scm/${params.scm_project}/"
 
 	def config
+	def configObj
```
Comment (suryajak) [inline jenkins-build-pack-website/Jenkinsfile]: Similar comments as in delete-serverless-service-build-pack/Jenkinsfile

File: jazz-build-module/service-metadata-loader.groovy (hunk 1 above)
Comment (SiniWilson) [reply jazz-build-module/service-metadata-loader.groovy]: Not clear

Comment (suryajak) [reply jazz-build-module/service-metadata-loader.groovy]: The method should me **loadServiceMetaData**(_serviceId_)

Comment (SiniWilson) [reply jazz-build-module/service-metadata-loader.groovy]: The loadServiceMetaData() method is calling /platform/services with Query Params. The loadServiceMetaDataByServiceId() method is calling /platform/services/{service_id}. The response also different. Thats why kept two separate methods.

File: delete-serverless-service-build-pack/Jenkinsfile (hunk 2 above)
Comment (SiniWilson) [reply delete-serverless-service-build-pack/Jenkinsfile]: In cooperated review comments

File: jazz-build-module/service-metadata-loader.groovy (hunk 1 above)
Comment (SiniWilson) [reply jazz-build-module/service-metadata-loader.groovy]: In cooperated review comments

File: jenkins-build-pack-api/Jenkinsfile (hunk 4 above)
Comment (SiniWilson) [reply jenkins-build-pack-api/Jenkinsfile]: In cooperated review comments

**Comment by SiniWilson** [comment]
In co operated the review Comments

**Pull request merged**